sdk/lib/opentelemetry/sdk/trace/span.rb

```diff
@@ -351,7 +351,7 @@ def truncate_attribute_values(attrs)
           return EMPTY_ATTRIBUTES if attrs.nil?
 
           attribute_length_limit = @span_limits.attribute_length_limit
-          attrs.each { |key, value| attrs[key] = OpenTelemetry::Common::Utilities.truncate(value, attribute_length_limit) } if attribute_length_limit
+          attrs.each { |key, value| attrs[key] = OpenTelemetry::Common::Utilities.truncate_attribute_value(value, attribute_length_limit) } if attribute_length_limit && value.is_a?(String)
```

Comment:
I think it's correct as is if we remove the `&& value.is_a?(String)`
```suggestion
          attrs.each { |key, value| attrs[key] = OpenTelemetry::Common::Utilities.truncate_attribute_value(value, attribute_length_limit) } if attribute_length_limit
```
We type check inside `truncate_attribute_value`. It might be slightly more efficient to check before the call 🤷‍♂️ .

Reply:
Oh wait @fbogsany , hehe, I mixed up my two PRs. I think you are right as well.